### PR TITLE
Python 3 fixes

### PR DESCRIPTION
**Pull request by patricksnape**
This branch passes all tests using Python 3 (on OSX). The major changes were:

  - as_bytes -> as_str_any
    - coremltools wants strings for keys and so we make sure they are strings rather than abusing the bytes are strings property from Python 2
  - xrange is not supported on Python 3
  - force integer division where expected
  - Fix absolute imports by refactoring the layers module to create a common layers module that is shared by the specific shape layers

### CODE REVIEW

**Comment by TobyRoseman** [comment]
These changes look good to me.

File: tfcoreml/_layers.py [hunk 1]
```diff
@@ -1017,8 +989,8 @@ def argmax(op, context):
 
 def extract_image_patches(op, context):
   # use a big convolution layer (that has weights!) for this op
-  input_name = compat.as_bytes(op.inputs[0].name)
-  output_name = compat.as_bytes(op.outputs[0].name)
+  input_name = compat.as_str_any(op.inputs[0].name)
+  output_name = compat.as_str_any(op.outputs[0].name)
   ksizes = op.get_attr('ksizes')
   padding_type = op.get_attr('padding')
```
Comment (aseemw) [inline tfcoreml/_layers.py]: This should also be changed to:
`padding_type = compat.as_str_any(op.get_attr('padding'))`

**Comment by aseemw** [comment]
@patricksnape Great!! I'll test again and merge. 
I'm not sure about the six version. We have a dependency on coremltools, which pins it to 1.10, @TobyRoseman do you know why?

**Comment by TobyRoseman** [comment]
@aseemw - I don't know why we pin that version.

**Comment by jonas-arkulpa** [comment]
Hey guys,

we've an issue which seems related to this.
I'm able to convert my model with Python 2.7 but not with 3.6.

The following error occurs if I try it with Python 3:
```shell
Traceback (most recent call last):
  File "tf2coreml.py", line 15, in <module>
    image_scale = 1 / 255.0)
  File ".../tf-coreml/tfcoreml/_tf_coreml_converter.py", line 483, in convert
    predicted_probabilities_output=predicted_probabilities_output)
  File ".../tf-coreml/tfcoreml/_tf_coreml_converter.py", line 285, in _convert_pb_to_mlmodel
    optimize_nn_spec(builder=builder)
  File ".../tf-coreml/tfcoreml/optimizations/_optimize_nn_spec.py", line 44, in optimize_nn_spec
    _optimize_fold_load_constants(builder.nn_spec.layers)
  File ".../tf-coreml/tfcoreml/optimizations/_optimize_nn_spec.py", line 12, in _optimize_fold_load_constants
    _optimize._fold_constants(nn_layers)
  File ".../tf-coreml/tfcoreml/optimizations/_optimize.py", line 187, in _fold_constants
    _replace_with_load_constant(nn_layers, i, y, shape, load_constant_outputs)
  File ".../tf-coreml/tfcoreml/optimizations/_optimize.py", line 124, in _replace_with_load_constant
    nn_layers[ind].loadConstant.MergeFromString('')
TypeError: a bytes-like object is required, not 'str'
```
System: Ubuntu 17.10
Build from current master: https://github.com/tf-coreml/tf-coreml/commit/e8db7a836c1c8164ebdc9e37738db8e00cf2537f
with: pip install -e .

Does this seem related or should I open another issue?

Many thanks in advance.

**Comment by patricksnape** [comment]
Yes I would open another issue - probably with the contents of your script - or at least the call to tfcoreml

**Pull request merged**